### DISCUSSION TITLE
Installing EBUSD on Debain 12, unmet dependencies Libssl 3

### DISCUSSION THREAD

**Discussion by Jomps86** [john30/ebusd#1263] [Q&A]
Hi all,

i need help installing ebusd on the debain of my raspberry 4. The aim is to integrate it in IOBroker, so I followed the video of "verdrahtet.info" (https://www.youtube.com/embed/5X_OUs1aTls) and the Readme (https://github.com/john30/ebusd-debian/blob/master/README.md).

When trying to install ebusd with
`apt-get install ebusd`

I get the following statement:
`Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:
The following packages have unmet dependencies:
 ebusd : Depends: libssl3 (>= 3.0.0) but it is not installable
E: Unable to correct problems, you have held broken packages.`

I have tried different things to install the libssl3, but i don´t get it... example
`sudo apt-get openssl-devel`
I have also tried using aptitude, but this does not work as well.

Yes I am a newbie, but know a bit coding and mechatronics...
Hardware btw it the eBUS Adapter Shield v5 and it is connected with USB to the Raspberry.

I hope that you can help me... as I am not sure if this is due to my raspberry or the installation...

Thank´s and best

**Comment by john30** [comment]
which libssl version does your system have?
EDIT: and did you pick the right file from ebusd-debian?

**Reply by Jomps86** [reply]
Hi John,

the Version of the opensll:
```
Package: openssl
Version: 1.1.1w-0+deb11u1+rpt1
Priority: optional
Section: utils
Maintainer: Debian OpenSSL Team <pkg-openssl-devel@lists.alioth.debian.org>

Installed-Size: 1,483 kB
Depends: libc6 (>= 2.17), libssl1.1 (>= 1.1.1)
```


So it seems to be the missing Depentds Libssl3. I´ll tried to get it but I fail. It tells me that Libssl has no installing candidate, so I tried to follow this link:
https://askubuntu.com/questions/1414956/libssl3-3-0-2-0ubuntu1-2-required-but-only-3-0-2-0ubuntu1-1-is-available-on-t

But it does not work...
I am a bit lost...

**Reply by Jomps86** [reply]
I resolved it. Problem was that the repository URLs where not correct.
For documentation if anyone has simlilar issues: 
Open the `/etc/apt/sources.list.` 
For my Debain 12 the URLs to inclulde where:
```
deb http://deb.debian.org/debian/ bookworm main contrib non-free
deb http://deb.debian.org/debian/ bookworm-updates main contrib non-free
deb http://security.debian.org/debian-security bookworm-security main contrib non-free
```
Then do an update and now you can install the Libssl3 with:
`apt install libssl3`
After that I was able to install the ebusd.
